development/core/src/SODaS/SODaSEntry.cpp

```diff
@@ -674,6 +674,8 @@ SODaSNameSpaceRoot* SODaSEntry::getNameSpaceRoot(
 		return NULL;
 	}
 
+	SOCmnSingleLock<SOCmnSync> lock(&g_engineSync);
```

Comment:
@mkentie Why must this lock be acquired here ? The only other place where it is used is in "SODaSEntry* getSODaSEntry(void) (line 122)" to guard the global g_engine.